### PR TITLE
Create rpm workflow

### PR DESCRIPTION
**Pull request by hubertqc**
Create a GitHub workflow to build RPM package to install the SELInux policy module, each time a new release is created.

### CODE REVIEW

File: .github/workflows/build_rpm.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          body: |
+            Changes in this Release
+            - Create RPM
+            - Upload Source RPM
+          draft: false
+          prerelease: false
+
+      - name: Replace Tokens in RPM spec file
+        run: |
+            #latest_rlse_url="https://api.github.com/repos/hubertqc/selinux_springboot/releases/latest"
```
Comment (Laurent-Gaillard) [inline .github/workflows/build_rpm.yml:41]: Is this the syntax or is it a comment?

File: rpm/selinux_springboot.spec [hunk 2]
```diff
@@ -0,0 +1,52 @@
+Name:		springboot-selinux
+Version:	#{version}#
+Release:        1%{?dist}
```
Comment (Laurent-Gaillard) [inline rpm/selinux_springboot.spec:3]: How do you want to name it?

File: rpm/selinux_springboot.spec [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+assigned the springboot_exec_t SELinux type.
+The Springboot application will run in the springboot_t domain.
+
+%prep
+%setup -q -n selinux_springboot-%{version}
+
+%build
+
+mkdir -p -m 0755 %{buildroot}/usr/share/selinux/packages/targeted
+mkdir -p -m 0755 %{buildroot}/usr/share/selinux/devel/include/contrib
```
Comment (Laurent-Gaillard) [inline rpm/selinux_springboot.spec:29]: Maybe more in application sub directory. Contrib seems to be use by system delivered or base policy delivered apps.

File: rpm/selinux_springboot.spec [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+mv springboot.pp.bz2 /usr/share/selinux/packages/targeted/
+cd /tmp
+rm -rf /tmp/selinux-springboot
+
+%postun
+semodule -r springboot
+
+%files
+/usr/share/selinux/devel/include/contrib/springboot.if
+%verify(not size filedigest mtime) /usr/share/selinux/packages/targeted/springboot.pp.bz2
```
Comment (Laurent-Gaillard) [inline rpm/selinux_springboot.spec:49]: What is it for?

**Comment by Laurent-Gaillard** [comment]
Maybe an idea to have a look https://github.blog/2022-02-10-using-reusable-workflows-github-actions/ in the future.
Like a repo to build all other repos rpms.
Not sure we can do it with our free accounts. 
Didn't have a deeper look at it for today. Maybe tomorrow 😋

**Comment by Laurent-Gaillard** [comment]
À little commit squash?

**Review by Laurent-Gaillard — DISMISSED** [review]

File: .github/workflows/build_rpm.yml (hunk 1 above)
Comment (hubertqc) [reply .github/workflows/build_rpm.yml:41]: It's a comment

File: rpm/selinux_springboot.spec (hunk 4 above)
Comment (hubertqc) [reply rpm/selinux_springboot.spec:49]: Before the RPM is installed, this BZ2 file is a tarball of the SELinux module source code.
After the post-install scriptlet, the file is BZ2 of the compiled PP module file.

Therefore, RPM-Verify should not care about this file contents, checksum, size,... to validate the package.

File: rpm/selinux_springboot.spec (hunk 3 above)
Comment (hubertqc) [reply rpm/selinux_springboot.spec:29]: I had done that at first, but the directory contains one solitary file...

File: rpm/selinux_springboot.spec (hunk 2 above)
Comment (hubertqc) [reply rpm/selinux_springboot.spec:3]: Not sure yet if I want the distro in the RPM release name...

File: rpm/selinux_springboot.spec (hunk 3 above)
Comment (Laurent-Gaillard) [reply rpm/selinux_springboot.spec:29]: image
As an example

Comment (Laurent-Gaillard) [reply rpm/selinux_springboot.spec:29]: that's why I proposed that.

Comment (hubertqc) [reply rpm/selinux_springboot.spec]: Ok for apps subir

**Review by Laurent-Gaillard — APPROVED** [review]

**Comment by hubertqc** [comment]
Create a GitHub Workflow so that a new release is created and a RPM pack is built when a `v*` tag is pushed.

**Pull request merged**